DOCSP-43090 - Authentication (#57)
mongodb/docs-c: source/security/authentication.txt

.. _c-auth:

Authentication Mechanisms
=========================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference
 
.. meta::
   :keywords: authorize, ecs, amazon

Overview
--------

> This guide describes the mechanisms you can use in the {+driver-short+} to authenticate
> users.

Reviewer: ```suggestion
This guide describes the mechanisms you can use in the {+driver-short+} to authenticate
users to MongoDB.
```

.. important:: Percent-Encoding

   You must :wikipedia:`percent-encode <Percent-encoding>` a username and password before
   you include them in a MongoDB URI.

.. _c-scram-sha-256:

SCRAM-SHA-256
-------------

SCRAM-SHA-256, as defined by `RFC 7677 <https://www.rfc-editor.org/rfc/rfc7677>`__,
is the default authentication mechanism on MongoDB deployments
running MongoDB v4.0 or later.

> To authenticate with this mechanism, set the following connection options:

Reviewer: ```suggestion
To authenticate by using SCRAM-SHA-256, set the following connection options:
```

- ``username``: The username to authenticate. Percent-encode this value before including
  it in the connection URI.
- ``password``: The password to authenticate. Percent-encode this value before including
  it in the connection URI.
- ``authSource``: The MongoDB database to authenticate against. By default,
  the {+driver-short+} authenticates against the database in the connection

>   URI, if you include one. If you don't, it authenticates against the ``admin`` database.

Reviewer: ```suggestion
  URI, if you include one. If you don't specify an authentication database,
  it authenticates against the ``admin`` database.
```

Reviewer: applies to all sections

Author: didn't find any more instances, but fixed the one here
